### PR TITLE
add org.kde.kstars

### PR DESCRIPTION
**Pull request by cho2**
WIP

### CODE REVIEW

**Comment by cho2** [comment]
bot, build org.kde.kstars

**Comment by cho2** [comment]
bot, build org.kde.kstars

**Comment by cho2** [comment]
bot, build org.kde.kstars

File: org.kde.kstars.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+    "id": "org.kde.kstars",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "kstars",
+    "rename-icon": "kstars",
+    "finish-args": [
+        "--share=ipc", 
+        "--socket=x11",
```
Comment (barthalion) [inline org.kde.kstars.json:10]:
```suggestion
        "--socket=fallback-x11", 
```

File: org.kde.kstars.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        "--share=ipc", 
+        "--socket=x11", 
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network"
+    ],
+
+    "modules": [
+        {
+            "name": "eigen",
```
Comment (barthalion) [inline org.kde.kstars.json:18]: You should cleanup static libraries and development headers.

**Review by barthalion — CHANGES_REQUESTED** [review]

File: org.kde.kstars.json (hunk 1 above)
Comment (cho2) [reply org.kde.kstars.json]: https://github.com/flathub/flathub/pull/1831/commits/7a539377beb221a14ef5e64f4bf55b1277b7966c

**Comment by eszlari** [comment]
I already submitted this one: https://github.com/flathub/flathub/pull/641 (blocked for technical reasons)

**Comment by nedrichards** [comment]
bot, build org.kde.kstars

**Comment by cho2** [comment]
bot, build org.kde.kstars

**Comment by cho2** [comment]
> I already submitted this one: #641 (blocked for technical reasons)

It's two years since the [first pull request](https://github.com/flathub/flathub/pull/641). Look like the problem is in udev. I think [udev already in kde runtime](https://invent.kde.org/packaging/flatpak-kde-runtime/-/commits/qt5.15lts?utf8=%E2%9C%93&search=udev) nowadays.

**Comment by cho2** [comment]
cc @flathub/kde

File: appdata.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+--- a/org.kde.kstars.appdata.xml	2020-07-18 14:39:01.000000000 +0700
++++ b/org.kde.kstars.appdata.xml	2020-09-15 11:54:12.639581651 +0700
+@@ -470,4 +470,8 @@
+     <value key="KDE::windows_store">https://www.microsoft.com/store/apps/9pprz2qhlxtg</value>
+     <value key="KDE::google_play">https://play.google.com/store/apps/details?id=org.kde.kstars.lite</value>
+   </custom>
++  <releases>
```
Comment (aleixpol) [inline appdata.patch:7]: Is this patch upstream yet?

Comment (cho2) [reply appdata.patch:7]: https://invent.kde.org/education/kstars/-/merge_requests/80 only for `content_rating`

**Comment by tsdgeos** [comment]
Seems reasonable, we have

-- The following RUNTIME packages have not been found:

 * Xplanet, Renders an image of all the major planets and most satellites, <http://xplanet.sourceforge.net>
   Gives KStars support for xplanet.
 * AstrometryNet, Astrometrics Library, <http://www.astrometry.net>
   Support for plate solving in KStars.

-- The following OPTIONAL packages have not been found:

 * CFitsio, FITS IO Library, <https://heasarc.gsfc.nasa.gov/fitsio/fitsio.html>
   Support for the FITS (Flexible Image Transport System) data format in KStars.
 * INDI (required version >= 1.7.1), Astronomical instrumentation control, <https://www.indilib.org>
   Support for controlling astronomical devices on Linux with KStars.
 * Nova
 * LibRaw, Library for reading RAW files, <https://www.libraw.org/>
   Support for reading and displaying RAW files in KStars.
 * WCSLIB, World Coordinate System library, <https://www.atnf.csiro.au/people/mcalabre/WCS>
   WCS enables KStars to read and process world coordinate systems in FITS header.
 * Qt5DataVisualization
 * Qt5Keychain


I think it'd be good if we can enable some of those, but we can do it at a later stage if you prefer

**Comment by eszlari** [comment]
quote from upstream (@knro):

> Absolutely. If udev rules cannot be somehow exported, then this is a no-go. It would be a support nightmare once the questions starting pouring in about why isn't this working as expected. I would love for this work, but the current limitation is unfortunately a show stopper.

https://github.com/flathub/flathub/pull/641#issuecomment-424974801

**Comment by knro** [comment]
Does flatpak now supports udev rules? If not, then nothing changes and we would have a KStars version stripped from a very important feature (INDI and Ekos).

**Comment by tsdgeos** [comment]
Probably not.

https://github.com/flatpak/flatpak/issues/961

@alexlarsson Can you confirm there's no plan to support udev still?

**Comment by aleixpol** [comment]
```
$ find /usr/lib* -name *udev*
/usr/lib/udev
/usr/lib/systemd/system/systemd-udevd.service.d
/usr/lib/x86_64-linux-gnu/libudev.so.1.6.18
/usr/lib/x86_64-linux-gnu/libudev.so.1
```

There is some udev stuff going on, I have no idea about rules though...

**Comment by cho2** [comment]
bot, build org.kde.kstars

**Comment by cho2** [comment]
bot, build org.kde.kstars

**Comment by cho2** [comment]
bot, build org.kde.kstars

**Comment by petrushy** [comment]
Hi, I have a somewhat stalled work on getting a kstars flatpak, some of the items there can probably be useful:
https://github.com/petrushy/org.kde.kstars/blob/master/org.kde.kstars.yaml

I think it was the USB stuff that I got stuck on.

**Pull request closed without merging**